ios/CodePush/CodePushPackage.m

```diff
@@ -391,16 +418,26 @@ + (NSDictionary *)getPackage:(NSString *)packageHash
     
     NSString *updateMetadataString = [NSString stringWithContentsOfFile:updateMetadataFilePath
                                                                encoding:NSUTF8StringEncoding
-                                                                  error:error];
+                                                                  error:&localError];
     
-    if (*error) {
+    if (!updateMetadataString) {
+        if (localError && error) {
+            *error = localError;
+        }
         return nil;
     }
     
     NSData *updateMetadata = [updateMetadataString dataUsingEncoding:NSUTF8StringEncoding];
-    return [NSJSONSerialization JSONObjectWithData:updateMetadata
+    id result = [NSJSONSerialization JSONObjectWithData:updateMetadata
                                            options:kNilOptions
-                                             error:error];
+                                             error:&localError];
+
+    if (!result) {
```

Comment:
If this fails, wouldn't you want the caller of this method to know that it was a failure within `NSJSONSerialization` ?